test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -1960,6 +1960,21 @@ BOOST_AUTO_TEST_CASE(balance)
 	ABI_CHECK(callContractFunction("getBalance()"), encodeArgs(23));
 }
 
+BOOST_AUTO_TEST_CASE(blocknumber)
+{
+	char const* sourceCode = R"(
+		contract test {
+			function getBlockNumber() public view returns (uint256 blockNumber) {
+				blockNumber = block.number;
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+	u256 startBlock = m_blockNumber;
+	for (u256 block = startBlock; block < startBlock + 10; block++)
+		ABI_CHECK(callContractFunction("getBlockNumber()"), encodeArgs(block));
```

Comment:
This fails with previous versions of aleth, for which we get ``block + 1`` instead of ``block``.

Reply:
I think this is the key test to consider. ``m_blockNumber`` is the block number of the transaction receipt of the last transaction and the same as calling ``eth_blockNumber``, which is specified as "number of most recent block". However, ``block.number`` should return the "current block number".

``compileAndRun`` mines the block containing the deployment of the contract. ``startBlock`` should be the block number of this block. Each call to ``callContractFunction`` will mine a new block containing the call.

So I would still expect to get ``encodeArgs(block + 1)`` here, as it used to be for older versions of ``aleth``, instead of ``encodeArgs(block)``. I'm not sure what has changed here (@gumb0?).